### PR TITLE
add bandit

### PR DESCRIPTION
**Pull request by bdwyer2**

### CODE REVIEW

File: core/mask_rcnn/mrcnn/utils.py [hunk 1]
```diff
@@ -121,7 +121,8 @@ def non_max_suppression(boxes, scores, threshold):
     scores: 1-D array of box scores.
     threshold: Float. IoU threshold to use for filtering.
     """
-    assert boxes.shape[0] > 0
+    if boxes.shape[0] < 0:
```
Comment (bdwyer2) [inline core/mask_rcnn/mrcnn/utils.py:124]:
```suggestion
    if boxes.shape[0] <= 0:
```

File: core/mask_rcnn/mrcnn/model.py [hunk 2]
```diff
@@ -27,8 +27,10 @@
 
 # Requires TensorFlow 1.3+ and Keras 2.0.8+.
 from distutils.version import LooseVersion
-assert LooseVersion(tf.__version__) >= LooseVersion("1.3")
-assert LooseVersion(keras.__version__) >= LooseVersion('2.0.8')
+if LooseVersion(tf.__version__) < LooseVersion("1.3"):
+    raise AssertionError("invalid tensorflow version: %r" % tf.__version__)
```
Comment (xuhdev) [inline core/mask_rcnn/mrcnn/model.py:31]:
```suggestion
    raise RuntimeError("invalid tensorflow version: %r" % tf.__version__)
```

File: core/mask_rcnn/mrcnn/model.py [hunk 3]
```diff
@@ -27,8 +27,10 @@
...
+if LooseVersion(keras.__version__) < LooseVersion('2.0.8'):
+    raise AssertionError("invalid keras version: %r" % keras.__version__)
```
Comment (xuhdev) [inline core/mask_rcnn/mrcnn/model.py:33]:
```suggestion
    raise RuntimeError("invalid keras version: %r" % keras.__version__)
```

File: core/mask_rcnn/mrcnn/model.py [hunk 4]
```diff
@@ -1254,8 +1258,10 @@ def hook(images, augmenter, parents, default):
         mask = det.augment_image(mask.astype(np.uint8),
                                  hooks=imgaug.HooksImages(activator=hook))
         # Verify that shapes didn't change
-        assert image.shape == image_shape, "Augmentation shouldn't change image size"
-        assert mask.shape == mask_shape, "Augmentation shouldn't change mask size"
```
Comment (xuhdev) [inline core/mask_rcnn/mrcnn/model.py:1258]: This might be a valid use of `assert`?

File: core/mask_rcnn/mrcnn/model.py [hunk 5]
```diff
@@ -1307,20 +1313,22 @@ def build_detection_targets(rpn_rois, gt_class_ids, gt_boxes, gt_masks, config):
...
-    assert gt_class_ids.dtype == np.int32, "Expected int but got {}".format(
-        gt_class_ids.dtype)
-    assert gt_boxes.dtype == np.int32, "Expected int but got {}".format(
-        gt_boxes.dtype)
-    assert gt_masks.dtype == np.bool_, "Expected bool but got {}".format(
-        gt_masks.dtype)
+    if rpn_rois.shape[0] <= 0:
+        raise ValueError("rpn_rois.shape[0] <= 0")
+    if gt_class_ids.dtype != np.int32:
+        raise ValueError("Expected int but got {}".format(gt_class_ids.dtype))
```
Comment (xuhdev) [inline core/mask_rcnn/mrcnn/model.py:1319]:
```suggestion
        raise TypeError("Expected int but got {}".format(gt_class_ids.dtype))
```

File: core/mask_rcnn/mrcnn/model.py [hunk 6]
```diff
@@ -1307,20 +1313,22 @@ def build_detection_targets(rpn_rois, gt_class_ids, gt_boxes, gt_masks, config):
...
+    if gt_boxes.dtype != np.int32:
+        raise ValueError("Expected int but got {}".format(gt_boxes.dtype))
```
Comment (xuhdev) [inline core/mask_rcnn/mrcnn/model.py:1321]:
```suggestion
        raise TypeError("Expected int but got {}".format(gt_boxes.dtype))
```

File: core/mask_rcnn/mrcnn/model.py [hunk 7]
```diff
@@ -1307,20 +1313,22 @@ def build_detection_targets(rpn_rois, gt_class_ids, gt_boxes, gt_masks, config):
...
+    if gt_masks.dtype != np.bool_:
+        raise ValueError("Expected bool but got {}".format(gt_masks.dtype))
```
Comment (xuhdev) [inline core/mask_rcnn/mrcnn/model.py:1323]:
```suggestion
        raise TypeError("Expected bool but got {}".format(gt_masks.dtype))
```

File: core/mask_rcnn/mrcnn/model.py (hunk 4 above)
Comment (bdwyer2) [reply core/mask_rcnn/mrcnn/model.py:1258]: Agreed, do you think it's best to leave it as it was and add a `# nosec` comment?

Comment (feihugis) [reply core/mask_rcnn/mrcnn/model.py:1258]: Or, we can raise an error as other changes if the check fails. Either works for me.

Comment (xuhdev) [reply core/mask_rcnn/mrcnn/model.py:1258]: A `#nosec` should work

**Pull request merged**